packages/snaps-sdk/src/types/handlers/name-lookup.ts

```diff
@@ -15,23 +15,45 @@ export type DomainLookupArgs = BaseOnNameLookupArgs & {
 };
 
 /**
- * The result of a domain lookup.
+ * The arguments for an address lookup.
+ *
+ * @property address - The address that is to be resolved.
+ */
+export type AddressLookupArgs = BaseOnNameLookupArgs & {
+  address: string;
+  domain?: never;
+};
+
+/**
+ * The address resolution object.
  *
+ * @property protocol - The protocol that resolved the domain.
  * @property resolvedAddress - The resolved address.
  */
-export type DomainLookupResult = {
+export type AddressResolution = {
```

Comment:
We should probably assert that the response of the name lookup snap matches these types, that will make the integration easier:

We can add it to: https://github.com/MetaMask/snaps/blob/main/packages/snaps-controllers/src/snaps/SnapController.ts#L2887C10-L2887C36